source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm

```diff
@@ -357,6 +374,80 @@ - (void)processImageConcurrently:(std::shared_ptr<Image> const &)imageElem
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0),
+            ^{
+                NSString *urlStr = [NSString stringWithCString:act->GetIconUrl().c_str() encoding:[NSString defaultCStringEncoding]];
+                // generate key for imageMap from image element's id
+                NSString *key = [NSString stringWithCString:act->GetId().c_str() encoding:[NSString defaultCStringEncoding]];
+                NSURL *url = [NSURL URLWithString:urlStr];
+                
+                // download image
+                UIImage *img = [UIImage imageWithData:[NSData dataWithContentsOfURL:url]];
+                
+                // UITask can't be run on global queue, add task to main queue
+                dispatch_async(dispatch_get_main_queue(),
+                    ^{
+                        __block UIButton *button = nil;
+                        // syncronize access to image map
+                        dispatch_sync(_serial_queue,
+                            ^{
+                                if(!_actionsMap[key]) {// UIButton is not ready, cache UIImage
+                                    _actionsMap[key] = img;
+                                } else {// UIButton ready, get view
+                                    button = _actionsMap[key];
+                                }
+                            });
+                        
+                        // if view is available, set image to it, and continue image processing
+                        if(button)
+                        {
+                            // Format the image so it fits in the button and is placed where it must be placed
+                            CGSize contentSize = [button.titleLabel intrinsicContentSize];
+                            double imageHeight = contentSize.height;
+                            CGSize originalImageSize = [img size];
+                            double scaleRatio = imageHeight / originalImageSize.height;
+                            double imageWidth = scaleRatio * originalImageSize.width;
+                            
+                            UIImage *scaledImage = [self scaleImage:img toSize:CGSizeMake(imageWidth, imageHeight)];
+                            [button setImage:scaledImage forState:UIControlStateNormal];
+                            
+                            IconPlacement iconPlacement = [_hostConfig getHostConfig]->actions.iconPlacement;
+                            if( iconPlacement == AdaptiveCards::IconPlacement::AboveTitle ){
+                                // Changes the insets for image, title and contents so the icon is rendered above the title
+                                CGFloat totalHeight = (imageHeight + contentSize.height);
+                                [button setImageEdgeInsets:UIEdgeInsetsMake(-(totalHeight - 2 * imageHeight), 0.0f, 0.0f, -contentSize.width)];
```

Comment:
isn't this zero? since totalHeight = imgeHeight(=contentSize.height) + contentSize.height, so totalHeight == 2 * imageHeight